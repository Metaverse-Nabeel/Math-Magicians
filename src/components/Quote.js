// eslint-disable-next-line import/no-extraneous-dependencies
import { ProgressBar } from 'react-loader-spinner';
import React, { useState, useEffect } from 'react';

const Quote = () => {
  const [loader, setLoader] = useState(false);
  const [hasError, setHasError] = useState(false);
  const [quotes, setQuotes] = useState({});
  const url = 'https://api.api-ninjas.com/v1/quotes?category=knowledge';
  const fetchData = async () => {
    setLoader(true);
    try {
      const res = await fetch(url, {
        method: 'GET',
        headers: {
          'X-Api-Key': 'nMRVJ4zVR4EMNn1c9g+SUQ==a1aZuCMIEnxLwKvD',
        },
      });
      const json = await res.json();
      setQuotes(json);
    } catch (error) {
      setHasError(true);
    }
    setLoader(false);
  };
  useEffect(() => {
    fetchData();
  }, []);
  if (loader) return <ProgressBar />;
  if (hasError) return <div className="errorAPI">Sorry! No Quotes right now. Try again later.</div>;
  return (
    <div className="quotesBody">
      <h2 className="quotesCategory">{quotes[0]?.category}</h2>
      <p className="quoteStatement">{`"${quotes[0]?.quote}"`}</p>
      <p className="quoteAuthor">{quotes[0]?.author}</p>
    </div>
  );
};

export default Quote;
