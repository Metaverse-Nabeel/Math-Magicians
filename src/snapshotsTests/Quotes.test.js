import React from 'react';
import { render } from '@testing-library/react';
import Quote from '../components/Quote';

describe('Tests for the Quotes page', () => {
  it('Should render the Quotes page properly', () => {
    const quotesPage = render(<Quote />);
    expect(quotesPage).toMatchSnapshot();
  });
});
