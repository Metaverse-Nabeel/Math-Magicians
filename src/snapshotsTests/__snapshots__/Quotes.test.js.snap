// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the Quotes page Should render the Quotes page properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        aria-busy="true"
        aria-label="progress-bar-loading"
        class=""
        data-testid="progress-bar-svg"
        height="80"
        preserveAspectRatio="xMidYMid"
        role="status"
        viewBox="0 0 100 100"
        width="80"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <clippath
            height="100"
            id="lds-progress-cpid-5009611b8a418"
            ng-attr-id="{{config.cpid}}"
            width="100"
            x="0"
            y="0"
          >
            <rect
              height="100"
              width="66.6667"
              x="0"
              y="0"
            >
              <animate
                attributeName="width"
                begin="0s"
                calcMode="linear"
                dur="1"
                keyTimes="0;0.5;1"
                repeatCount="indefinite"
                values="0;100;100"
              />
              <animate
                attributeName="x"
                begin="0s"
                calcMode="linear"
                dur="1"
                keyTimes="0;0.5;1"
                repeatCount="indefinite"
                values="0;0;100"
              />
            </rect>
          </clippath>
        </defs>
        <path
          d="M82,63H18c-7.2,0-13-5.8-13-13v0c0-7.2,5.8-13,13-13h64c7.2,0,13,5.8,13,13v0C95,57.2,89.2,63,82,63z"
          fill="none"
          ng-attr-stroke="{{config.c1}}"
          stroke="#F4442E"
          stroke-width="2.7928"
        />
        <path
          clip-path="url(#lds-progress-cpid-5009611b8a418)"
          d="M81.3,58.7H18.7c-4.8,0-8.7-3.9-8.7-8.7v0c0-4.8,3.9-8.7,8.7-8.7h62.7c4.8,0,8.7,3.9,8.7,8.7v0C90,54.8,86.1,58.7,81.3,58.7z"
          fill="#51E5FF"
          ng-attr-clip-path="url(#{{config.cpid}})"
          ng-attr-fill="{{config.c2}}"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      aria-busy="true"
      aria-label="progress-bar-loading"
      class=""
      data-testid="progress-bar-svg"
      height="80"
      preserveAspectRatio="xMidYMid"
      role="status"
      viewBox="0 0 100 100"
      width="80"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <clippath
          height="100"
          id="lds-progress-cpid-5009611b8a418"
          ng-attr-id="{{config.cpid}}"
          width="100"
          x="0"
          y="0"
        >
          <rect
            height="100"
            width="66.6667"
            x="0"
            y="0"
          >
            <animate
              attributeName="width"
              begin="0s"
              calcMode="linear"
              dur="1"
              keyTimes="0;0.5;1"
              repeatCount="indefinite"
              values="0;100;100"
            />
            <animate
              attributeName="x"
              begin="0s"
              calcMode="linear"
              dur="1"
              keyTimes="0;0.5;1"
              repeatCount="indefinite"
              values="0;0;100"
            />
          </rect>
        </clippath>
      </defs>
      <path
        d="M82,63H18c-7.2,0-13-5.8-13-13v0c0-7.2,5.8-13,13-13h64c7.2,0,13,5.8,13,13v0C95,57.2,89.2,63,82,63z"
        fill="none"
        ng-attr-stroke="{{config.c1}}"
        stroke="#F4442E"
        stroke-width="2.7928"
      />
      <path
        clip-path="url(#lds-progress-cpid-5009611b8a418)"
        d="M81.3,58.7H18.7c-4.8,0-8.7-3.9-8.7-8.7v0c0-4.8,3.9-8.7,8.7-8.7h62.7c4.8,0,8.7,3.9,8.7,8.7v0C90,54.8,86.1,58.7,81.3,58.7z"
        fill="#51E5FF"
        ng-attr-clip-path="url(#{{config.cpid}})"
        ng-attr-fill="{{config.c2}}"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
